actionpack/lib/action_dispatch/middleware/exception_wrapper.rb

```diff
@@ -111,6 +124,16 @@ def original_exception(exception)
         end
       end
 
+      def wrapped_causes_for(exception, backtrace_cleaner)
+        causes = []
+
+        while cause = (cause || exception).cause
+          causes << self.class.new(backtrace_cleaner, cause)
+        end
+
+        causes
+      end
```

Comment:
I was wondering if there's a better way to convert a linked list to an `Enumerable`. Open to suggestions,

Reply:
I was trying to fiddle around with Enumerator but couldn't really come up with something.

Reply:
Maybe? (untested)

```ruby
def causes_for(exception)
  return enum_for(__method__, exception) unless block_given?
  yield exception while exception = exception.cause
end

def wrapped_causes_for(exception, backtrace_cleaner)
  causes_for(exception).map { |cause| self.class.new(backtrace_cleaner, cause) }
end
```

Reply:
@matthewd tried it in my console, it works!